File: examples/benchmarks/pi_leibniz.py
import time

def calculate_pi(iterations):
    pi = 0.0
    sign = 1.0
    
    for k in range(iterations):
        pi += sign / (2.0 * k + 1.0)
        sign = -sign
    
    return pi * 4.0

if __name__ == "__main__":
    iterations = 100000000
    
    start = time.perf_counter()
    pi = calculate_pi(iterations)
    duration = time.perf_counter() - start
    
    print(f"π ≈ {pi:.10f}")
    print(f"Iterations: {iterations}")
    print(f"Time: {duration:.6f} seconds")

File: examples/benchmarks/pi_leibniz.nim
import times

proc calculate_pi(iterations: int): float =
    var pi = 0.0
    var sign = 1.0
    
    for k in 0..<iterations:
        pi += sign / (2.0 * float(k) + 1.0)
        sign = -sign
    
    pi * 4.0

when isMainModule:
    let iterations = 100_000_000
    
    let start = cpuTime()
    let pi = calculate_pi(iterations)
    let duration = cpuTime() - start
    
    echo "π ≈ ", pi
    echo "Iterations: ", iterations
    echo "Time: ", duration, " seconds"

